barretenberg/cpp/src/barretenberg/transcript/transcript.hpp

```diff
@@ -385,6 +385,13 @@ template <typename TranscriptParams> class BaseTranscript {
             manifest.add_challenge(round_number, labels...);
         }
 
+        // In case the transcript is used for recursive verification, we need to sanitize current round data so we don't
+        // get an origin tag violation inside the hasher
```

Comment:
Can you elaborate a little more on why this is necessary?

Reply:
Added